_resources/port1.0/group/haskell_cabal-1.0.tcl

```diff
@@ -19,77 +19,124 @@
 # haskell_cabal.bin
 #   The cabal binary. Defaults to ${prefix}/bin/cabal.
 #
-# haskell_cabal.default_args
-#   Default arguments for cabal used across invocations in configure, build, and
-#   destroot phases. Defaults to
-#       --prefix=${prefix}
-#       --bindir=${prefix}/bin
-#       --libdir=${prefix}/lib
-#       --libsubdir=${name}
-#       --dynlibdir=${prefix}/lib
-#       --libexecdir=${prefix}/libexec
-#       --libexecsubdir=${name}
-#       --datadir=${prefix}/share/${name}
-#       --docdir=${prefix}/share/doc/${name}
-#       --htmldir=${prefix}/share/doc/${name}
-#       --sysconfdir=${prefix}/etc/${name}
-#       --enable-documentation
-#       --enable-relocatable
+# haskell_cabal.cabal_root
+#   The root path for cabal. Defaults to ${workpath}/.home/.cabal .
+#
+# haskell_cabal.gcc_location
+#   The path for gcc. Defaults to /usr/bin/gcc.
 
 proc haskell_cabal.add_dependencies {} {
     global name
     if { ${name} ne "cabal" } {
-        depends_build-append port:cabal \
-                             port:ghc
+        depends_build-append \
+            port:bash \
+            port:cabal \
+            port:ghc
     }
 }
 port::register_callback haskell_cabal.add_dependencies
 
+options haskell_cabal.cabal_root
+default haskell_cabal.cabal_root {${workpath}/.home/.cabal}
+
+options haskell_cabal.gcc_location
```

Comment:
to be consistent should probably just refer to this as 'cc_location' as the compiler will pretty much always be clang, not gcc.

Reply:
Thanks. I just removed the entire option and stuck this into the `config` file. For posterity, not that `cabal`'s `gcc-location` setting is it's name.
```
program-locations
  gcc-location: ${configure.cc}
```